### PR TITLE
Int 548 epic implement referral hoi backend

### PR DESCRIPTION
**Pull request by AmanShurafa**
### Jira Story

- [Adjust history card (HoI) to retrieve data from Legacy via new Ferb API endpoint, not Elasticsearch INT-528](https://osi-cwds.atlassian.net/browse/INT-528)

Pull requests related to this one:  

1. Point HOI to new Ferb API end-point:
PR: https://github.com/ca-cwds/intake/pull/439
Story: https://osi-cwds.atlassian.net/browse/INT-549

2. Referrals HOI
PR: https://github.com/ca-cwds/intake/pull/440
Story: https://osi-cwds.atlassian.net/browse/INT-549

3. Cases HOI
PR: https://github.com/ca-cwds/intake/pull/444
Story: https://osi-cwds.atlassian.net/browse/INT-550

4. Screenings HOI
PR: https://github.com/ca-cwds/intake/pull/448
Story: https://osi-cwds.atlassian.net/browse/INT-547
  
Passed CI test phase:
https://ci.mycasebook.org/job/intake_test_phase/job/int_548_epic_implement_referral_hoi_backend/

### CODE REVIEW

File: app/javascript/selectors/screening/historyOfInvolvementSelectors.js [hunk 1]
```diff
@@ -46,30 +45,30 @@ export const getFormattedReferralsSelector = createSelector(
   getReferralsSelector,
   (referrals) => referrals.map((referral) => {
     const status = referral.get('end_date') ? 'Closed' : 'Open'
-    const responseTime = referral.get('response_time')
-    const limitedAccessCode = referral.getIn(['access_limitation', 'limited_access_code'], 'N')
+    const responseTime = referral.getIn(['response_time', 'description'])
```
Comment (saraOSI) [inline app/javascript/selectors/screening/historyOfInvolvementSelectors.js]: I don't think we can rely on the response time object to have the description we need for Intake. I know the description was added to the API, but we should really be looking up the value (based on the system code) from the LOV service.

File: app/javascript/selectors/screening/historyOfInvolvementSelectors.js [hunk 2]
```diff
@@ -90,12 +89,12 @@ export const getFormattedScreeningsSelector = createSelector(
...
     })
     return fromJS({
-      county: COUNTIES[screening.get('county_name')] || '',
+      county: screening.getIn(['county', 'description'], ''),
       dateRange: dateRangeFormatter(screening.toJS()),
       people: notJustReporters.map((person) => nameFormatter(person.toJS())).join(', '),
       reporter: nameFormatter({name_default: '', ...screening.get('reporter', Map()).toJS()}),
       status: screening.get('end_date') ? 'Closed' : 'In Progress',
-      worker: screening.getIn(['assigned_social_worker', 'last_name'], ''),
+      worker: nameFormatter({name_default: '', ...screening.get('assigned_social_worker', Map()).toJS()}),
```
Comment (saraOSI) [inline app/javascript/selectors/screening/historyOfInvolvementSelectors.js]: Is there a reason to `.toJS()` this? Will `fromJS` blow up if part of the payload is already immutable?

Comment (AmanShurafa) [reply app/javascript/selectors/screening/historyOfInvolvementSelectors.js]: You are right. This is not needed

File: app/javascript/selectors/screening/historyOfInvolvementSelectors.js (hunk 1 above)
Comment (AmanShurafa) [reply app/javascript/selectors/screening/historyOfInvolvementSelectors.js]: Agreed

Comment (AmanShurafa) [reply app/javascript/selectors/screening/historyOfInvolvementSelectors.js]: Done

File: spec/javascripts/containers/HistoryTableContainerSpec.jsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+import React from 'react'
+import {createMockStore} from 'redux-test-utils'
+import {fromJS} from 'immutable'
+import {shallow} from 'enzyme'
+import * as IntakeConfig from 'common/config'
+
+describe('HistoryTableContainer', () => {
+  const state = fromJS({
+    involvements: {
+      referrals: [
```
Comment (anuroop15) [inline spec/javascripts/containers/HistoryTableContainerSpec.jsx]: doesn't HOI container spec missing cases and screening ??

File: spec/javascripts/containers/HistoryTableContainerSpec.jsx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+  let component
+  beforeEach(() => {
+    const context = {store}
+    spyOn(IntakeConfig, 'isFeatureActive').and.returnValue(true)
+    component = shallow(<HistoryTableContainer />, {context})
+  })
+
+  it('passes formatted referrals HistoryTable view', () => {
+    expect(component.find('HistoryTable').props().referrals).toEqual([{
+      dateRange: '12/02/2017 - 12/02/2017',
```
Comment (anuroop15) [inline spec/javascripts/containers/HistoryTableContainerSpec.jsx]: specs are missing for cases and screening

File: spec/javascripts/containers/HistoryTableContainerSpec.jsx (hunk 3 above)
Comment (AmanShurafa) [reply spec/javascripts/containers/HistoryTableContainerSpec.jsx]: Yes it does in the final version of the file.

File: spec/javascripts/containers/HistoryTableContainerSpec.jsx (hunk 4 above)
Comment (AmanShurafa) [reply spec/javascripts/containers/HistoryTableContainerSpec.jsx]: They exist in the final version of the file.

File: spec/javascripts/containers/HistoryTableContainerSpec.jsx (hunk 3 above)
Comment (anuroop15) [reply spec/javascripts/containers/HistoryTableContainerSpec.jsx]: ok!!

File: app/javascript/selectors/screening/historyOfInvolvementSelectors.js [hunk 5]
```diff
@@ -44,32 +44,34 @@ const getReferralsSelector = createSelector(
...
 export const getFormattedReferralsSelector = createSelector(
   getReferralsSelector,
-  (referrals) => referrals.map((referral) => {
+  getScreenResponseTimesSelector,
+  (referrals, responseTimes) => referrals.map((referral) => {
     const status = referral.get('end_date') ? 'Closed' : 'Open'
-    const responseTime = referral.get('response_time')
-    const limitedAccessCode = referral.getIn(['access_limitation', 'limited_access_code'], 'N')
+    const responseTimeID = referral.getIn(['response_time', 'id'])
+    const responseTime = systemCodeDisplayValue(responseTimeID, responseTimes)
```
Comment (saraOSI) [inline app/javascript/selectors/screening/historyOfInvolvementSelectors.js]: 👍

**Pull request merged**